From 5d89eb01c93d8a62998e3bdccae28a7732e3bd51 Mon Sep 17 00:00:00 2001
From: Gagan Mac <gmac@codeaurora.org>
Date: Thu, 28 Feb 2013 13:25:33 -0700
Subject: [PATCH] msm: msm_bus: Add NULL pointer check

This patch protects the bus driver against clients who try to
update requests without registering the client.

Change-Id: Ie740844218dbabe5fca098798c461590aba42e97
CRs-Fixed: 459651
Signed-off-by: Gagan Mac <gmac@codeaurora.org>
---
 arch/arm/mach-msm/msm_bus/msm_bus_arb.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-msm/msm_bus/msm_bus_arb.c b/arch/arm/mach-msm/msm_bus/msm_bus_arb.c
index e5616b9..6af2a56 100644
--- a/arch/arm/mach-msm/msm_bus/msm_bus_arb.c
+++ b/arch/arm/mach-msm/msm_bus/msm_bus_arb.c
@@ -562,7 +562,7 @@ int msm_bus_scale_client_update_request(uint32_t cl, unsigned index)
 	int pnode, src, curr, ctx;
 	uint64_t req_clk, req_bw, curr_clk, curr_bw;
 	struct msm_bus_client *client = (struct msm_bus_client *)cl;
-	if (IS_ERR(client)) {
+	if (IS_ERR_OR_NULL(client)) {
 		MSM_BUS_ERR("msm_bus_scale_client update req error %d\n",
 				(uint32_t)client);
 		return -ENXIO;
-- 
1.7.8.3

